### DISCUSSION TITLE
Does it work with Vite?

### DISCUSSION THREAD

**Discussion by rimzzlabs** [wojtekmaj/react-pdf#1166] [Q&A]
Hi, Im using React with Vite, but it seems it doesn't work with vite, I got this error:
![Screen Shot 2022-11-15 at 14 54 10](https://user-images.githubusercontent.com/62492410/201861147-c9741c18-8a97-41fc-ae35-1993ab0f9032.png)

is it because Im using `React.lazy` for my component?

**Comment by rimzzlabs** [comment]
> Hi, Im using React with Vite, but it seems it doesn't work with vite, I got this error: ![Screen Shot 2022-11-15 at 14 54 10](https://user-images.githubusercontent.com/62492410/201861147-c9741c18-8a97-41fc-ae35-1993ab0f9032.png)
> 
> is it because Im using `React.lazy` for my component?

but if i remove the `<Document />` component from `react-pdf` it works fine

**Comment by rimzzlabs** [comment]
> > Hi, Im using React with Vite, but it seems it doesn't work with vite, I got this error: ![Screen Shot 2022-11-15 at 14 54 10](https://user-images.githubusercontent.com/62492410/201861147-c9741c18-8a97-41fc-ae35-1993ab0f9032.png)
> > is it because Im using `React.lazy` for my component?
> 
> but if i remove the `<Document />` component from `react-pdf` it works fine

![Screen Shot 2022-11-15 at 15 07 43](https://user-images.githubusercontent.com/62492410/201864804-cd60744a-61bc-4eda-bd3e-8d4db465a0f1.png)

okay so I just missed the sample of vite, I saw it and the import statement is by using vite esm, but it didn't work for me, because Im using typescript, it looks like the type definition for this hasn't been updated yet.

im using `@types/react-pdf^5.7.4`

**Reply by rimzzlabs** [reply]
> > > Hi, Im using React with Vite, but it seems it doesn't work with vite, I got this error: ![Screen Shot 2022-11-15 at 14 54 10](https://user-images.githubusercontent.com/62492410/201861147-c9741c18-8a97-41fc-ae35-1993ab0f9032.png)
> > > is it because Im using `React.lazy` for my component?
> > 
> > 
> > but if i remove the `<Document />` component from `react-pdf` it works fine
> 
> ![Screen Shot 2022-11-15 at 15 07 43](https://user-images.githubusercontent.com/62492410/201864804-cd60744a-61bc-4eda-bd3e-8d4db465a0f1.png)
> 
> okay so I just missed the sample of vite, I saw it and the import statement is by using vite esm, but it didn't work for me, because Im using typescript, it looks like the type definition for this hasn't been updated yet.
> 
> im using `@types/react-pdf^5.7.4`


I also didn't receive entry for vite inside node_modules

![Screen Shot 2022-11-15 at 15 25 06](https://user-images.githubusercontent.com/62492410/201867832-83ec443e-a5db-4af3-a4f0-01825e338fb0.png)

**Comment by pietrobondioli** [comment]
It seems that an PR on [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped) repository has already been opened to add type support for react-pdf v6 (which includes vite support). So to bypass the typescript error you have 2 options:

- Reflect type declaration changes of the [PR](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/63389/files) on you own code (node_module/@types/react-pdf/).
- Just declare the vite path as a module on your code:

![image](https://user-images.githubusercontent.com/55422938/205466149-c41f203f-82c7-4aec-a407-8f7563cec334.png)
![image](https://user-images.githubusercontent.com/55422938/205466134-b4923dec-31f8-4fea-ac1e-114f5e25e392.png)
 and on your tsconfig.json add to compiler options:

`
"typeRoots": [
            "node_modules/@types",
            "src/@types"
        ]
`

then the error will be gone and react-pdf on vite will still be working as expected:

![image](https://user-images.githubusercontent.com/55422938/205466280-5281baed-91bb-404b-ac96-dd08a008c1cf.png)


Hope it helps.
